app/code/Magento/InventoryCatalogSearch/Plugin/Search/FilterMapper/AdaptStockStatusFilterPlugin.php

```diff
@@ -158,15 +168,25 @@ private function addSubProductEntityJoin(Select $select, $mainTableAlias)
      */
     private function addInventoryStockJoin(Select $select, $showOutOfStockFlag)
     {
-        $select->joinInner(
+        $select->joinLeft(
             ['stock_index' => $this->getStockTableName()],
             'stock_index.sku = product.sku',
             []
         );
         if ($showOutOfStockFlag === false) {
+            $globalManageStock = (int)$this->stockConfiguration->getManageStock();
+            $stockSettingsCondition = '(legacy_stock_item.use_config_manage_stock = 0 AND legacy_stock_item.manage_stock = 0)';
+            if (0 === $globalManageStock) {
+                $stockSettingsCondition .= ' OR legacy_stock_item.use_config_manage_stock = 1';
+            }
```

Comment:
You can use here \Magento\InventorySales\Model\ResourceModel\IsStockItemSalableCondition\ManageStockCondition